tests/helpers.mjs

```diff
@@ -1,77 +1,62 @@
 import assert from 'node:assert';
-import { join } from 'node:path';
 
 import { beforeAll, afterAll } from 'vitest';
 import { sync as resolveBinSync } from 'resolve-bin';
 import { execa } from 'execa';
 import tmp from 'tmp-promise';
 
-import copyWithTemplate from '../lib/copy-with-template';
+import copyWithTemplate from '../lib/copy-with-template.js';
+import { newApp } from '../lib/new-app.mjs';
 
 function findEmber() {
   return resolveBinSync('ember-cli', { executable: 'ember' });
 }
 
 export const emberCli = findEmber();
 
-const blueprintPath = join(__dirname, '..');
 const appName = 'fancy-app-in-test';
 
 export function newProjectWithFixtures({
   flags = [],
   fixturePath,
   name = appName,
 } = {}) {
-  let tmpDir;
+  let info;
 
   assert(fixturePath, `a fixturePath is required`);
 
   beforeAll(async () => {
-    tmpDir = await tmp.dir({ unsafeCleanup: true });
-
-    let emberCliArgs = [
-      'new',
-      name,
-      '-b',
-      blueprintPath,
-      '--pnpm',
-      '--skip-git',
-      ...flags,
-    ];
-
-    await execa(emberCli, emberCliArgs, {
-      cwd: tmpDir.path,
-      preferLocal: true,
-    });
+    info = await newApp({ name, flags: ['--skip-git', ...flags] });
 
     // apply the fixture on top of the generated app
-    copyWithTemplate(join(__dirname, 'fixture'), join(tmpDir.path, name), {
+    copyWithTemplate(fixturePath, info.dir, {
       name,
     });
 
     // Sync the lints for the fixtures with the project's config
     await execa(`pnpm`, ['lint:fix'], {
-      cwd: join(tmpDir.path, name),
+      cwd: info.dir,
     });
   });
 
   afterAll(async () => {
     try {
-      await tmpDir.cleanup();
+      await tmp.cleanup();
     } catch {
       // if it fails to cleaup we don't want to break CI
     }
   });
 
   return {
```

Comment:
this structure got cleaned up quite a bit because the joins and all that are now in new-app